### ISSUE TITLE
wrong output for ls with newlines in filename

### ISSUE THREAD

**Issue by JensTimmerman** [kellyjonbrazil/jc#26]
![image](https://user-images.githubusercontent.com/281523/74935475-99701900-53e8-11ea-87a1-55b4d6d51c98.png)

` jc ls -lah` results in : `"Error - ls parser could not parse the input data. Did you use the correct parser?`

**Comment by kellyjonbrazil** [comment]
I'm not sure there will be a great solution for this since `ls` does not seem to encode the newline when the output is piped.

With standard `ls` it will always look like multiple files and I don't see a way to detect that it's really just one file.

With `ls -l` we can at least see that something is amiss. Instead of failing like it does now, I could do a number of things:

- simply ignore parsing errors and keep going
- detect the error and add a key to the output called `corrupted_filename` or something, so you'd have something that looks like:
```
{
    "filename": "file with newline in the name",
    "corrupted_filename": true
    "flags": "-rw-rw-r--.",
    "links": 1,
    "owner": "joe",
    "group": "joe",
    "size": 0,
    "date": "Feb 13 17:52"
  },
```
- Do the above but try to piece the filename back together by stitching together the trailing newline text. 

The second option seems reasonable to me. Does that make sense or do you think there is a better way to approach this?

Thanks,
kb

**Comment by kellyjonbrazil** [comment]
I have a fix for `ls -l` that rebuilds the filename with one or more encoded `\n`s:
```
$ jc -p ls -l
[
  {
    "filename": "another filename with\nmultiple\nnewline characters",
    "flags": "-rw-r--r--",
    "links": 1,
    "owner": "kbrazil",
    "group": "wheel",
    "size": 0,
    "date": "Feb 24 14:56"
  },
  {
    "filename": "filename with a\nnewline character",
    "flags": "-rw-r--r--",
    "links": 1,
    "owner": "kbrazil",
    "group": "wheel",
    "size": 0,
    "date": "Feb 24 14:55"
  },
  {
    "filename": "regularfile",
    "flags": "-rw-r--r--",
    "links": 1,
    "owner": "kbrazil",
    "group": "wheel",
    "size": 0,
    "date": "Feb 24 14:56"
  }
]
```
Note that since regular `ls` without `-l` doesn't provide any distinction between a filename with newlines and multiple files, so it will still come back incorrect:
```
$ jc -p ls
[
  {
    "filename": "another filename with"
  },
  {
    "filename": "multiple"
  },
  {
    "filename": "newline characters"
  },
  {
    "filename": "filename with a"
  },
  {
    "filename": "newline character"
  },
  {
    "filename": "regularfile"
  }
]
```
https://github.com/kellyjonbrazil/jc/commit/964868c8aff99edf37b0db41c7d16f8b84ac4704

**Comment by kellyjonbrazil** [comment]
Added some fixes for multiple consecutive newline characters and one or more trailing newlines in a filename. Commit here: https://github.com/kellyjonbrazil/jc/commit/116e07f1614b4a45eb58ffcfe20b5efa71c473a8

**Comment by kellyjonbrazil** [comment]
More fixes for filenames that start with newlines: https://github.com/kellyjonbrazil/jc/commit/573b27946474276592ee7494689ce9a88f5a05f7

**Comment by kellyjonbrazil** [comment]
I’ll work on unit tests and should have this fix into v1.7.5 in the next couple days.
